### PR TITLE
Fix/separate sitemap cli and restructure

### PR DESCRIPTION
**Pull request by BarathaAberathne**
### What

Rearranged the folder structure and separated the cli tool from main sitemap tool. Added build-cli command into make file to provide a separate builder to the tool.

### How to review

Check if the folder structuring change is applied successfully. 

### Who can review

@lindenmckenzie @dionisislolas

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -17,6 +17,10 @@ audit:
 build:
 	go build -tags 'production' $(LDFLAGS) -o $(BINPATH)/dp-sitemap
 
+.PHONY: build-cli
+build-cli:
+	go build -tags 'production' $(LDFLAGS) -o $(BINPATH)/dp-sitemap-cli cmd/cli-tool/*.go
+
```
Comment (lindenmckenzie) [inline Makefile:23]: would suggest to have two commands - one for local running and one for remote. Different tags and GOOS etc. 

Include this in the README for the CLI tool as well.

File: cmd/cli-tool/utilities/utilities.go [hunk 2]
```diff
@@ -23,6 +23,10 @@ func createCliSitemapGenerator(cfg *config.Config, commandline *FlagFields) (*si
 
 	var transport http.RoundTripper = dphttp.DefaultTransport
 
+	if !commandline.FakeScroll {
+		cfg.OpenSearchConfig.Signer = true
+		fmt.Println("Signer set true")
+	}
```
Comment (lindenmckenzie) [inline cmd/cli-tool/utilities/utilities.go]: This seems to counteract the other PR?

File: main.go [hunk 3]
```diff
@@ -43,7 +38,7 @@ func run(ctx context.Context) error {
 	signals := make(chan os.Signal, 1)
 	signal.Notify(signals, os.Interrupt, syscall.SIGTERM)
 
-	robotseo.Init(cmd.CmdFlagFields.RobotsFilePathReader)
+	robotseo.Init(utilities.CmdFlagFields.RobotsFilePathReader)
```
Comment (lindenmckenzie) [inline main.go]: This is still conflating the app and CLI implementation.

**Review by lindenmckenzie — CHANGES_REQUESTED** [review]
Few comments.

File: cmd/cli-tool/utilities/utilities.go (hunk 2 above)
Comment (BarathaAberathne) [reply cmd/cli-tool/utilities/utilities.go]: Yes, will remove this if block.

**Review by BarathaAberathne — COMMENTED** [review]
issues Addressed

File: cmd/cli-tool/README.md [hunk 4]
```diff
@@ -15,3 +15,12 @@ Set the opensearch signer value using "export OPENSEARCH_SIGNER=true" when runni
       --sitemap-file-path-reader string   path to sitemap files that we are reading from (default "./sitemap/static/")
       --sitemap-index string              OPENSEARCH_SITEMAP_INDEX (default "1")
       --zebedee-url string                zebedee url (default "http://localhost:8082")
+
+# Build Commands
+
+To build for Remote envirionment (Sandbox/Prod..etc):
+      make build-cli-remote
+This will create a Linux build
```
Comment (lindenmckenzie) [inline cmd/cli-tool/README.md]: better to fence the command with backticks. e.g.

```sh
 make build-cli-remote
```

File: main.go [hunk 5]
```diff
@@ -43,8 +36,6 @@ func run(ctx context.Context) error {
 	signals := make(chan os.Signal, 1)
 	signal.Notify(signals, os.Interrupt, syscall.SIGTERM)
 
-	robotseo.Init(cmd.CmdFlagFields.RobotsFilePathReader)
-
 	// Run the service, providing an error channel for fatal errors
```
Comment (lindenmckenzie) [inline main.go]: You're still removing the init?

**Review by lindenmckenzie — COMMENTED** [review]
Couple comments.

File: main.go (hunk 5 above)
Comment (BarathaAberathne) [reply main.go]: Thought this will be added from dionisi's PR

Comment (BarathaAberathne) [reply main.go]: This line will be added again with next commit.

**Review by BarathaAberathne — COMMENTED** [review]
Comments addressed.

**Review by BarathaAberathne — COMMENTED** [review]
Comments Addressed

File: cmd/cli-tool/README.md [hunk 6]
```diff
@@ -15,3 +15,13 @@ Set the opensearch signer value using "export OPENSEARCH_SIGNER=true" when runni
       --sitemap-file-path-reader string   path to sitemap files that we are reading from (default "./sitemap/static/")
       --sitemap-index string              OPENSEARCH_SITEMAP_INDEX (default "1")
       --zebedee-url string                zebedee url (default "http://localhost:8082")
+
+# Build Commands
+
+To build for Remote envirionment (Sandbox/Prod..etc):
+      'make build-cli-remote'
```
Comment (lindenmckenzie) [inline cmd/cli-tool/README.md]:
```suggestion
      ```sh
         make build-cli-remote
      ```
```

File: cmd/cli-tool/main.go [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+	"context"
+
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+func main() {
+	ctx := context.Background()
+
+	cmdErr := GetRootCommand().Execute()
+
```
Comment (lindenmckenzie) [inline cmd/cli-tool/main.go]: Usually don't leave gaps after errors.

**Review by lindenmckenzie — COMMENTED** [review]
Couple of minor comments.

**Review by lindenmckenzie — APPROVED** [review]

**Pull request merged**